pkg/gui/confirmation_panel.go

```diff
@@ -13,133 +7,265 @@ import (
+// without any handlers.
+// currentView: is used to return focus.
+// title: what to call the view.
+// handleConfirm: what to do on confirmation.
+// returns an error if something goes wrong.
+func (gui *Gui) createPromptPanel(currentView *gocui.View, title string, handleConfirm handler) error {
+
+	_, err := gui.g.SetViewOnBottom("commitMessage")
+	if err != nil {
+		gui.Log.Errorf("Failed to set view on bottom at createPromptPanel: %s\n", err)
+		return err
+	}
+	confirmationView, err := gui.prepareConfirmationPanel(currentView, title, "")
+	if err != nil {
+		gui.Log.Errorf("Failed to prepare confirmation panel at createPromptPanel: %s\n", err)
+		return err
+	}
+
+	confirmationView.Editable = true
+
+	err = gui.setConfirmationHandlers(handleConfirm, nil)
+	if err != nil {
+		gui.Log.Errorf("Failed to setConfirmationHandlers at createPromptPanel: %s\n", err)
+		return err
+	}
+
+	return nil
+}
+
+// createErrorPanel creates a error panel which is just a confirmation panel.
+// message: the error to display.
+// returns an error if something goes wrong.
+func (gui *Gui) createErrorPanel(message string) error {
+
+	gui.Log.Error(message)
+
+	currentView := gui.g.CurrentView()
+
+	colorFunction := color.New(color.FgRed).SprintFunc()
+	coloredMessage := colorFunction(strings.TrimSpace(message))
+
+	err := gui.createConfirmationPanel(currentView, gui.Tr.SLocalize("Error"), coloredMessage, nil, nil)
+	if err != nil {
+		gui.Log.Errorf("Failed to create confirmation panel at createErrorPanel: %s\n", err)
+		return err
+	}
+
+	return nil
+}
+
+// createMessagePanel creates a message panel which is just a confirmation panel.
+// without any handlers.
+// currentView: is used to return focus.
+// title: what to call the view.
+// prompt: what to display.
+// returns an error if something goes wrong.
+func (gui *Gui) createMessagePanel(currentView *gocui.View, title, prompt string) error {
+	return gui.createConfirmationPanel(currentView, title, prompt, nil, nil)
+}
+
+// wrappedConfirmationFunction creates a function that is acceptable for some other functions.
+// TO BE REMOVED.
+func (gui *Gui) wrappedConfirmationFunction(function handler) handler {
+
+	// TODO change this
 	return func(g *gocui.Gui, v *gocui.View) error {
 		if function != nil {
-			if err := function(g, v); err != nil {
+			if err := function(gui.g, v); err != nil {
 				return err
 			}
 		}
-		return gui.closeConfirmationPrompt(g)
+		return gui.closeConfirmationPrompt()
 	}
+
 }
 
-func (gui *Gui) closeConfirmationPrompt(g *gocui.Gui) error {
-	view, err := g.View("confirmation")
+// closeConfirmationPrompt closes the confirmation prompt.
+// returns an error if something goes wrong.
+func (gui *Gui) closeConfirmationPrompt() error {
+
+	view, err := gui.g.View("confirmation")
+	if err != nil {
+		gui.Log.Errorf("Failed to get confirmation view at closeConfirmationPrompt: %s\n", err)
+		return err
+	}
+
+	err = gui.returnFocus(gui.g, view)
```

Comment:
we can remove gui.g from this

Reply:
Almost done with view_handlers.go then this will be removed